trunk/NDHMS/Routing/module_channel_routing.F

```diff
@@ -962,29 +982,41 @@ Subroutine drive_CHANNEL(latval,lonval,KT, IXRT,JXRT, SUBRTSWCRT, &
                end do ! do m
 #endif
                    
-                if(TYPEL(k) .eq. 1) then   !--link is a reservoir
-
-                   ! CALL LEVELPOOL(1,QLINK(k,1), Qup, QLINK(k,1), QLINK(k,2), &
-                   !  QLateral(k), DT, RESHT(k), HRZAREA(k), LAKEMAXH(k), &
-                   !  WEIRC(k), WEIRL(k),ORIFICEE(k),  ORIFICEC(k), ORIFICEA(k))
-
-                   elseif (channel_option .eq. 1) then  !muskingum routing
+                if(TYPEL(k) == 1) then   !--link is a reservoir
+                    l_idx = lake_lookup(k)
+                    if (l_idx >= 0) then     !-- -999 if not a reservoir in the lookup table (belt-and-suspenders check)
+                        print *, "CALL LEVELPOOL", l_idx,Quc, Qup, QLINK(k,2), QLateral(k), DT, &
+                                RESHT(l_idx), HRZAREA(l_idx), WEIRH(l_idx), LAKEMAXH(l_idx), &
+                                WEIRC(l_idx), WEIRL(l_idx), ORIFICEE(l_idx), ORIFICEC(l_idx), ORIFICEA(l_idx)
+                        CALL LEVELPOOL(l_idx,Quc, Qup, QLINK(k,2), QLateral(k), DT, &
```

Comment:
We have (potentially) divergent LP functions at this point.  I would recommend moving this to call the reservoir module levelpool function, and would actually deprecate/remove the channel routing module LEVELPOOL definition in favor of the reservoir module definition.

Reply:
This PR is for version 5.1.1 of the community WRF-Hydro code, in which there is no Reservoir module.

Reply:
Ah, I see.  Still an interesting conversation to be had about keeping that branch synced to some degree, but not relevant to this PR at this point in time.